### PR TITLE
Add roc chart to gallery

### PR DESCRIPTION
**Pull request by RossKen**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [X] DOC

### PR Checklist

- [ ] Added documentation for changes
- [ ] Added feature to example notebooks or tutorial (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/charts/roc_chart_from_labels_table.ipynb [hunk 1]
```diff
@@ -172,15 +169,39 @@
...
    "source": [
-    "### What the chart shows\n"
+    "### What the chart shows\n",
+    "\n",
+    "The chart plots the True Positive Rate against False Positive Rate for clerically reviewed records. Each point on the curve reflects the choice of a match weight threshold for a match and the subsequent True/False Positive Rates.\n",
+    "\n",
+    "??? note \"What the chart tooltip shows\"\n",
+    "    ![](./img/roc_chart_from_labels_table_tooltip.png)\n",
+    "\n",
+    "    The tooltip shows information based on the point on the curve that the user is hoverng over, including:\n",
```
Comment (ThomasHepworth) [inline docs/charts/roc_chart_from_labels_table.ipynb:179]: A quick aside, should we adjust the `truth_threshold` chart label? I understand why it's been named as such, but it's not the clearest description.

File: docs/charts/roc_chart_from_labels_table.ipynb [hunk 2]
```diff
@@ -172,15 +169,39 @@
...
+    "\n",
+    "    - The match weight and match probability threshold\n",
+    "    - The False and True Positive Rate\n",
+    "    - The count of True Positives, True Negatives, False Positives and False Negatives\n",
+    "    - Precision, Recall and F1 score\n",
```
Comment (ThomasHepworth) [inline docs/charts/roc_chart_from_labels_table.ipynb:184]: Is it worth linking to an article on precision, recall and f1?

**Review by ThomasHepworth — APPROVED** [review]
Looks great, thanks.

File: docs/charts/roc_chart_from_labels_table.ipynb (hunk 2 above)
Comment (RossKen) [reply docs/charts/roc_chart_from_labels_table.ipynb:184]: Added

**Pull request merged**